viper/src/verifier.rs

```diff
@@ -4,345 +4,14 @@
-                        self.jni
-                            .unwrap_result(verification_error_wrapper.call_fullId(viper_error)),
-                    );
-
-                    let pos = self
-                        .jni
-                        .unwrap_result(verification_error_wrapper.call_pos(viper_error));
-
-                    let message =
-                        self.jni.to_string(self.jni.unwrap_result(
-                            verification_error_wrapper.call_readableMessage(viper_error),
-                        ));
-
-                    let pos_id =
-                        if self
-                            .jni
-                            .is_instance_of(pos, "viper/silver/ast/HasIdentifier")
-                        {
-                            Some(self.jni.get_string(
-                                self.jni.unwrap_result(has_identifier_wrapper.call_id(pos)),
-                            ))
-                        } else {
-                            debug!(
-                                "The verifier returned an error whose position has no identifier: {:?}",
-                                self.jni.to_string(viper_error)
-                            );
-                            None
-                        };
-
-                    let offending_node = self
-                        .jni
-                        .unwrap_result(verification_error_wrapper.call_offendingNode(viper_error));
-
-                    let offending_pos = self
-                        .jni
-                        .unwrap_result(error_node_positioned_wrapper.call_pos(offending_node));
-
-                    let offending_pos_id =
-                        if self
-                            .jni
-                            .is_instance_of(offending_pos, "viper/silver/ast/HasIdentifier")
-                        {
-                            Some(self.jni.get_string(
-                                self.jni.unwrap_result(has_identifier_wrapper.call_id(offending_pos)),
-                            ))
-                        } else {
-                            debug!(
-                                "The verifier returned an error whose position has no identifier: {:?}",
-                                self.jni.to_string(viper_error)
-                            );
-                            None
-                        };
-
-                    errors.push(VerificationError::new(
-                        error_full_id,
-                        pos_id,
-                        offending_pos_id,
-                        reason_pos_id,
-                        message,
-                        counterexample,
-                    ))
-                }
-
-                VerificationResult::Failure(errors)
-            } else {
-                VerificationResult::Success
-            }
-        })
-    }
-}
+    fn verify(&mut self, program: crate::Program) -> VerificationResult;
```

Comment:
I would remove this method because this way, you could have the native backend completely independent of Viper.